metview 5.17.1 (new formula)

iainrussell opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

SMillerDev commented on Formula/metview.rb:
```diff
@@ -0,0 +1,87 @@
+  version "5.17.1"
+  sha256 "f9eb91918e2bc43388d53cb424de95fa352a3cb64f01cbd0dbd62303a05086f9"
+  license "Apache-2.0"
+
+  # NOTE: do not build on linux because of libtirpc problems - come back to it later
```
What problems?

iainrussell replied:
Our CMake detection of tirpc can require some tweaking, especially on Ubuntu, where the libs are not installed into /usr/lib by the normal package manager. I guess it might be simpler with brew, but I'd prefer to try it with a test system locally rather than try/fail/repeat with GitHub pushes. And macOS is our target system at the moment, so we'd like to get it available there before adding Linux support.

SMillerDev replied:
Unless the software is not meant to run on Linux I'd prefer not to ship half support in Homebrew.

iainrussell replied:
OK, let me try to get that working.

iainrussell replied:
Done, now working on Linux.

SMillerDev commented on Formula/metview.rb:
```diff
@@ -0,0 +1,87 @@
+  uses_from_macos "flex"  => :build
+
+  def install
+    mkdir "build" do
+      system "cmake", "..",
+                      "-DBUNDLE_SKIP_ECCODES=1",
+                      "-DECBUILD_LOG_LEVEL=DEBUG",
+                      "-DENABLE_MIR_DOWNLOAD_MASKS=OFF",
+                      "-DENABLE_BUILD_TOOLS=OFF",
+                      "-DENABLE_ECKIT_CMD=OFF",
+                      *std_cmake_args
+      system "make", "install"
+    end
```
```suggestion
    args = %w[
      "-DBUNDLE_SKIP_ECCODES=1",
      "-DENABLE_MIR_DOWNLOAD_MASKS=OFF",
      "-DENABLE_BUILD_TOOLS=OFF",
      "-DENABLE_ECKIT_CMD=OFF",
    ]
    system "cmake", "-S", ".", "-B", "build", *args, *cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

SMillerDev replied:
Did this modern CMake build format not work @iainrussell ?

iainrussell replied:
Done, thanks for the tip!

SMillerDev commented on Formula/metview.rb:
```diff
@@ -0,0 +1,87 @@
+      system "make", "install"
+    end
+
+    # Avoid references to Homebrew shims directory
+    # (ecbuild stores some references to the build directory - not used, so we can replace them)
```
if they're not used, can they be removed?

iainrussell replied:
Good point - we can probably remove most of them - is there a command I can run in the recipe to remove those files?

SMillerDev replied:
`rm "file"` or `rm_r "dir/"`

iainrussell replied:
Thanks, I've changed the formula to remove those files now - much better, as they are not needed!

SMillerDev commented on Formula/metview.rb:
```diff
@@ -0,0 +1,87 @@
+      system "make", "install"
+    end
+
+    # Avoid references to Homebrew shims directory
+    # (ecbuild stores some references to the build directory - not used, so we can remove them)
+    shim_references = [lib/"metview-bundle/include/mars_client_ecbuild_config.h",
+                       lib/"metview-bundle/include/eckit/eckit_ecbuild_config.h",
+                       lib/"metview-bundle/include/atlas/atlas_ecbuild_config.h",
+                       lib/"metview-bundle/include/mir/mir_ecbuild_config.h",
+                       lib/"metview-bundle/lib/pkgconfig/eckit.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_option.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_web.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_geometry.pc",
+                       lib/"metview-bundle/lib/pkgconfig/atlas-c++.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_test_value_custom_params.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_linalg.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_sql.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_maths.pc",
+                       lib/"metview-bundle/lib/pkgconfig/eckit_mpi.pc",
+                       lib/"metview-bundle/lib/pkgconfig/atlas.pc",
+                       lib/"metview-bundle/bin/metview_bin/compile"]
+    rm shim_references
```
If these are not needed, can we avoid them using CMake flags?

iainrussell replied:
Good question, I just checked and unfortunately not.

iainrussell replied:
I've simplified the removal of these files, the entire directories can in fact be removed.

iainrussell commented:
Hi @SMillerDev , I've simplified the things you suggested and got it working on Linux. I've also squashed the commits, so hopefully it's ready to go now.

SMillerDev reviewed (approved):
I'm fine with it being a follow up, but could you add a livecheck to the formula

iainrussell commented:
Sure, sounds like a good idea. Thanks @SMillerDev and @iMichka (and all at Brew)!

iainrussell commented:
Thanks everyone, it works great!
